### ISSUE TITLE
public ExcelRangeBase LoadFromCollection<T>(IEnumerable<T> Collection, bool PrintHeaders, TableStyles TableStyle, BindingFlags memberFlags, MemberInfo[] Members);

### ISSUE THREAD

**Issue by jhlovett** [EPPlusSoftware/EPPlus#185]
After version 5.2.1, there was a Bug in the LoadFromCollection method under ExcelRangeBase class. After passing in the MemberInfo[] Members parameter, the data was not loaded into Excel.

**Comment by swmal** [comment]
Thanks for reporting, we will fix this asap.

**Comment by swmal** [comment]
@jhlovett can you provide some code that replicates this bug? We have this unit test that works:

```csharp
internal abstract class BaseClass
{
    public string Id { get; set; }
    public string Name { get; set; }
}

internal class Implementation : BaseClass
{
    public int Number { get; set; }
}

[TestMethod]
public void ShouldFilterOneMember()
{
    var items = new List<BaseClass>()
    {
        new Implementation(){ Id = "123", Name = "Item 1", Number = 3}
    };
    using (var pck = new ExcelPackage(new MemoryStream()))
    {
        var sheet = pck.Workbook.Worksheets.Add("sheet");
        var t = typeof(Implementation);
        sheet.Cells["C1"].LoadFromCollection(items, true, TableStyles.Dark1, LoadFromCollectionParams.DefaultBindingFlags,
            new MemberInfo[]
            {
                t.GetProperty("Id"),
            });
        Assert.AreEqual("Id", sheet.Cells["C1"].Value);
        Assert.AreEqual("123", sheet.Cells["C2"].Value);
    }
}
```

Thanks.

**Comment by jhlovett** [comment]
`  public static MemoryStream ExportListToExcel<T>(IList<T> list, string hiddens)
        {
            if (list == null || list.Count == 0)
                throw new Exception("no data");
            using (ExcelPackage pck = new ExcelPackage())
            {
                ExcelWorksheet ws = pck.Workbook.Worksheets.Add("sheet1");
                PropertyInfo[] propertyInfos = LoadHeader<T>(ws, hiddens);
                if (propertyInfos == null || propertyInfos.Length == 0)
                {
                    throw new Exception("No export column");
                }

                ws.Cells["A2"].LoadFromCollection(list, false, OfficeOpenXml.Table.TableStyles.Custom, BindingFlags.Public, propertyInfos);

                for (int i = 0; i < propertyInfos.Length; i++)
                {
                    var prop = propertyInfos[i];
                    if (prop.PropertyType == typeof(DateTime) || prop.PropertyType == typeof(DateTime?))
                    {
                        ws.Column(i + 1).Style.Numberformat.Format = "yyyy-MM-dd HH:mm:ss";
                    }
                }

                using (ExcelRange rng = ws.Cells[1, 1, 1, propertyInfos.Length])
                {
                    rng.Style.Font.Bold = true;
                    rng.Style.Fill.PatternType = ExcelFillStyle.Solid;
                    rng.Style.Fill.BackgroundColor.SetColor(Color.FromArgb(79, 129, 189));
                    rng.Style.Font.Color.SetColor(Color.White);
                }

                var fileStream = new MemoryStream();
                pck.SaveAs(fileStream);
                fileStream.Position = 0;

                return fileStream;
            }
        }

  private static PropertyInfo[] LoadHeader<T>(ExcelWorksheet ws, string hiddens)
        {
            hiddens = string.IsNullOrEmpty(hiddens) ? "" : hiddens.Replace("\n", "").Replace(" ", "").Replace("\t", "").Replace("\r", "");
            var array = hiddens.Split(',').ToDictionary(x => x);

            PropertyInfo[] propertyInfos = typeof(T).GetProperties();
            var outputProperty = new List<PropertyInfo>();
            for (int i = 0; i < propertyInfos.Length; i++)
            {
                var property = propertyInfos[i];

                var attribute = (DisplayAttribute)property.GetCustomAttribute(typeof(DisplayAttribute));
                if (attribute != null && !array.ContainsKey(attribute.Name))
                {
                    outputProperty.Add(property);
                    ws.Cells[1, outputProperty.Count].Value = attribute.Name;
                }
            }

            return outputProperty.ToArray();
        }

`
entity:
` public class Test
    {
        /// 微信昵称
        /// </summary>
        [Display(Name = "昵称")]
        public string wx_nickName { get; set; }

        /// <summary>
        /// 姓名
        /// </summary>
        [Display(Name = "姓名")]
        public string Name { get; set; }
  }
`

call:
   public IActionResult Export(PayRechargeRecordSearchDto searchDto)
        {
         
            List<Test> list = new List<Test>() 
            {
                new Test(){ Name="张三",wx_nickName="zh"  },
                new Test(){ Name="李四",wx_nickName="li"  }
            };
            
            
            MemoryStream stream=ExcelHelper.ExportListToExcel(list, "");

            return File(stream, "application/octet-stream","充值流水.xlsx");
        }`
`
Versions 5.2.0 and below have data exported to Excel, and later versions have no data exported.

**Comment by swmal** [comment]
Thanks!

**Comment by swmal** [comment]
This will be fixed in the next version. In the meantime you can fix it by changing the call to LoadFromCollection to this:

```csharp
ws.Cells["A2"].LoadFromCollection(list, false, OfficeOpenXml.Table.TableStyles.Custom, BindingFlags.Public | BindingFlags.Instance, propertyInfos);
```

**Comment by JanKallman** [comment]
Fixed in version 5.3.2
